azurerm/internal/services/compute/resource_arm_dedicated_host_group.go

```diff
@@ -0,0 +1,203 @@
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroupName := d.Get("resource_group_name").(string)
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceGroupName, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for present of existing Dedicated Host Group %q (Resource Group %q): %+v", name, resourceGroupName, err)
+			}
+		}
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_dedicated_host_group", *existing.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	platformFaultDomainCount := d.Get("platform_fault_domain_count").(int)
+	t := d.Get("tags").(map[string]interface{})
+
+	parameters := compute.DedicatedHostGroup{
+		Location: utils.String(location),
+		DedicatedHostGroupProperties: &compute.DedicatedHostGroupProperties{
+			PlatformFaultDomainCount: utils.Int32(int32(platformFaultDomainCount)),
+		},
+		Tags: tags.Expand(t),
+	}
+	if zones, ok := d.GetOk("zones"); ok {
+		parameters.Zones = utils.ExpandStringSlice(zones.([]interface{}))
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceGroupName, name, parameters); err != nil {
+		return fmt.Errorf("Error creating Dedicated Host Group %q (Resource Group %q): %+v", name, resourceGroupName, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroupName, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Dedicated Host Group %q (Resource Group %q): %+v", name, resourceGroupName, err)
+	}
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read Dedicated Host Group %q (Resource Group %q) ID", name, resourceGroupName)
+	}
+	d.SetId(*resp.ID)
+
+	return resourceArmDedicatedHostGroupRead(d, meta)
+}
+
+func resourceArmDedicatedHostGroupRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Compute.DedicatedHostGroupsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroupName := id.ResourceGroup
+	name := id.Path["hostGroups"]
+
+	resp, err := client.Get(ctx, resourceGroupName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Dedicated Host Group %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading Dedicated Host Group %q (Resource Group %q): %+v", name, resourceGroupName, err)
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resourceGroupName)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	if props := resp.DedicatedHostGroupProperties; props != nil {
+		d.Set("platform_fault_domain_count", int(*props.PlatformFaultDomainCount))
```

Comment:
(as above) can we nil-check this and ensure we set a default value if this is nil here?